### PR TITLE
JUnit5 assertThrows MapRankingTest

### PR DESCRIPTION
**Pull request by nhojpatrick**

### CODE REVIEW

File: commons-math-neuralnet/src/test/java/org/apache/commons/math4/neuralnet/MapRankingTest.java [hunk 1]
```diff
@@ -94,15 +97,16 @@ public void testFindClosestNeuron() {
...
+    @Test
     public void testRankPrecondition() {
         final UniformRandomProvider rng = RandomSource.SPLIT_MIX_64.create();
         final FeatureInitializer init
             = new OffsetFeatureInitializer(FeatureInitializerFactory.uniform(rng, -0.1, 0.1));
         final FeatureInitializer[] initArray = {init};
 
-        new MapRanking(new NeuronString(3, false, initArray).getNetwork(),
-                       new EuclideanDistance()).rank(new double[] {-1}, 0);
+        assertThrows(IllegalArgumentException.class, () ->
```
Comment (aherbert) [inline commons-math-neuralnet/src/test/java/org/apache/commons/math4/neuralnet/MapRankingTest.java:107]: Here is an example where the source of the exception is not clear. Can you break this down to create the valid objects first and catch the exception from I presume `MapRanking.rank`.

**Comment by aherbert** [comment]
No need to force-push to bring in the other merged changes. I can rebase and merge. Can you address the isolation of the method that actually throws the exception which is the real benefit of using assertThrows.

File: commons-math-neuralnet/src/test/java/org/apache/commons/math4/neuralnet/MapRankingTest.java (hunk 1 above)
Comment (nhojpatrick) [reply commons-math-neuralnet/src/test/java/org/apache/commons/math4/neuralnet/MapRankingTest.java:111]: extracted setup, so only rank is checked.

**Comment by nhojpatrick** [comment]
I've a shell alias, so this is my default;
```
git fetch origin && git rebase origin/master && git push mine -f
```

**Pull request merged**